src/lib/core/cbus.c

```diff
@@ -433,7 +433,13 @@ cbus_call(struct cpipe *callee, struct cpipe *caller, struct cbus_call_msg *msg,
 
 	cpipe_push(callee, cmsg(msg));
 
-	fiber_yield_timeout(timeout);
+	if (timeout == TIMEOUT_INFINITY) {
+		fiber_set_cancellable(false);
```

Comment:
First, you don't restore the cancellable flag.

Second, conceptually this looks wrong. If you need a call that must not be cancelled, you should make your fiber non-cnancellable by yourself rather than patching cbus_call.

Reply:
Fixed